### PR TITLE
EXLM 483 ADLS Integration

### PR DESCRIPTION
**Pull request by devisree29**
EXLM 483 ADLS Integration
Instruction Led Courses for Curated Cards API Integration 

image

Jira ID:https://jira.corp.adobe.com/browse/EXLM-483

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.page/docs/integrations-learn/experience-cloud/solution-categories/content-management
- After: https://feature-exlm-483-adl--exlm--adobe-experience-league.hlx.live/en/curated-cards-adls

### CODE REVIEW

File: scripts/data-service/adls-data-service.js [hunk 1]
```diff
@@ -1 +1,39 @@
...
+  /**
+   * Fetches data from the configured ADLS Service.
+   *
+   * @returns {Promise<Array>} A promise that resolves with an array of data results.
+   */
+  async fetchDataFromSource() {
+    try {
+      if (ADLS in sessionStorage) {
+        return JSON.parse(sessionStorage[ADLS]);
+      }
```
[the remark this answers is not in the document]
Comment (devisree29) [reply scripts/data-service/adls-data-service.js]: Removed the logic for storing data in Session

File: scripts/browse-card/browse-card.js [hunk 2]
```diff
@@ -94,7 +98,12 @@ const buildCardContent = (card, model) => {
     const stringContent = description.length > 100 ? `${description.substring(0, 100).trim()}...` : description;
     const descriptionElement = document.createElement('p');
     descriptionElement.classList.add('browse-card-description-text');
-    descriptionElement.textContent = stringContent;
+    // Parse the HTML string
```
Comment (Sivaramvt) [inline scripts/browse-card/browse-card.js:101]: I suppose we should render the HTML as-is as it's trusted source
Per Noor, we remove the vulnerable tags and put it inside the <p> tag as innerHTML

File: scripts/browse-card/browse-card.js [hunk 3]
```diff
@@ -83,6 +87,10 @@ const buildCardCtaContent = ({ cardFooter, contentType, viewLink, viewLinkText }
   cardFooter.appendChild(anchorLink);
 };
 
+const stripScriptTags = function (input) {
+  return input.replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, '');
```
Comment (noormd08) [inline scripts/browse-card/browse-card.js:91]: Please convert to arrow function and directly use it.

**Review by noormd08 — APPROVED** [review]

File: scripts/browse-card/browse-card.js (hunk 2 above)
Comment (devisree29) [reply scripts/browse-card/browse-card.js]: Changed it according to the review comment

**Pull request merged**